client/app/bundles/course/assessment/submission/reducers/submissionFlags.js

```diff
@@ -13,15 +15,24 @@ const initialState = {
   isUnsubmitting: false,
   isDeleting: false,
   isSubmissionBlocked: false,
+  clientVersion: {},
+  savingStatus: {},
 };
 
 export default function (state = initialState, action) {
+  // eslint-disable-next-line sonarjs/max-switch-cases
   switch (action.type) {
     case actions.FETCH_SUBMISSIONS_REQUEST:
       return { ...state, isLoading: true };
     case actions.FETCH_SUBMISSION_SUCCESS:
-    case actions.FETCH_SUBMISSION_FAILURE:
-      return { ...state, isLoading: false };
+    case actions.FETCH_SUBMISSION_FAILURE: {
+      return produce(state, (draft) => {
+        draft.isLoading = false;
+        action.payload.answers.forEach((answer) => {
```

Comment:
When the API call fails, does your payload even have any content? You should just set everything inside draft.savingStatus to None, why do you need to get answer.id?

Reply:
You can try to attempt a new submission. Dont you encounter an error everytime?